### PR TITLE
[NDS-887] feat(Filter): implementation [part 2]

### PR DESCRIPTION
**Pull request by kostasdano**
## Description

[NDS-887](https://orfium.atlassian.net/browse/NDS-887)

[NDS-887]: https://orfium.atlassian.net/browse/NDS-887?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/DatePicker/DatePickInput/DatePickInput.tsx [hunk 1]
```diff
@@ -120,40 +113,22 @@ const DatePickInput = React.forwardRef<HTMLInputElement, DatePickInputProps>(
 
     const renderBase = () => {
```
Comment (panvourtsis) [inline src/components/DatePicker/DatePickInput/DatePickInput.tsx:114]: i feel that there is no need for `renderBase` anymore. We can just use `if` to return the end result

File: src/components/DatePicker/DatePicker.types.ts [hunk 2]
```diff
@@ -34,10 +34,12 @@ export type DatePickerProps = {
   isDefaultNow?: boolean;
   /** Style properties for the DatePicker with a filter base */
   filterConfig?: {
+    /** The filter's label */
+    label?: string;
     /** The type of the filter button's palette - defaults to "primary" */
-    buttonType?: 'primary' | 'secondary';
+    // buttonType?: 'primary' | 'secondary';
```
Comment (panvourtsis) [inline src/components/DatePicker/DatePicker.types.ts:40]: remove comments 🙏

File: src/components/Filter/components/FilterMenu/FilterMenu.style.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { css } from '@emotion/react';
+import type { Theme } from 'theme';
+import { rem } from 'theme/utils';
+
+/** @TODO: replace styles with menu tokens once Menu component is implemented for v5 */
```
Comment (panvourtsis) [inline src/components/Filter/components/FilterMenu/FilterMenu.style.ts:5]: this is done in `Menu`

File: src/components/Filter/components/FilterMenu/components/FilterSearchField/FilterSearchField.style.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { css } from '@emotion/react';
+import { rem } from 'theme/utils';
+
+export const textFieldWrapper = () =>
+  css`
+    border-bottom: 1px solid #b9cdfc;
```
Comment (panvourtsis) [inline src/components/Filter/components/FilterMenu/components/FilterSearchField/FilterSearchField.style.ts:6]: this doesnt have token?

File: src/components/Filter/components/FilterMenu/components/FilterSearchField/FilterSearchField.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  | 'selectedFilter'
+  | 'isMulti'
+  | 'isSearchable'
+  | 'isLoading'
+  | 'onClear'
+  | 'onFilterDelete'
+  | 'dataTestPrefixId'
+>;
+
+const FilterSearchField = ({
```
Comment (panvourtsis) [inline src/components/Filter/components/FilterMenu/components/FilterSearchField/FilterSearchField.tsx:27]: ref here?

File: src/components/Filter/components/FilterMenu/components/Options/Options.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+}) => {
+  const isForcedVirtualized = items.length > MAX_NON_VIRTUALIZED_ITEMS_FILTER;
+
+  const height = isForcedVirtualized ? FILTER_OPTIONS_MAX_HEIGHT : undefined;
+
+  const onSelectionChange = useCallback(
+    (keys: ListSelection) => {
+      const keyFound = String(head(Array.from(keys)));
+      if (keyFound === SELECT_ALL_OPTION.value) {
+        //@ts-ignore
```
Comment (panvourtsis) [inline src/components/Filter/components/FilterMenu/components/Options/Options.tsx:43]: `onChange` i feel that this is mistakenly placed here. The `onChange` should be the same as `List`. Therefore we must see over a technical discussion if needed how we can make it happen

File: src/components/Filter/constants.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
+export const options = [
```
Comment (panvourtsis) [inline src/components/Filter/constants.ts:1]: ❤️

**Review by panvourtsis — CHANGES_REQUESTED** [review]

File: src/components/Filter/components/FilterMenu/components/FilterSearchField/FilterSearchField.style.ts (hunk 4 above)
Comment (kostasdano) [reply src/components/Filter/components/FilterMenu/components/FilterSearchField/FilterSearchField.style.ts:6]: it does but its part of the search tokens. I will add a TODO comment to replace with search tokens when Search component is implemented.

**Review by panvourtsis — APPROVED** [review]

**Pull request merged**